interfaces/builtin/content_test.go

```diff
@@ -141,29 +139,109 @@ plugs:
 	c.Assert(err, ErrorMatches, "content interface target path is not clean:.*")
 }
 
-func (s *ContentSuite) TestConnectedPlugSnippetSimple(c *C) {
-	const mockSnapYaml = `name: content-slot-snap
-version: 1.0
+func (s *ContentSuite) TestResolveSpecialVariable(c *C) {
+	info := snaptest.MockInfo(c, "name: name", &snap.SideInfo{Revision: snap.R(42)})
+	c.Check(builtin.ResolveSpecialVariable("foo", info), Equals, "/snap/name/42/foo")
+	c.Check(builtin.ResolveSpecialVariable("$SNAP/foo", info), Equals, "/snap/name/42/foo")
+	c.Check(builtin.ResolveSpecialVariable("$SNAP_DATA/foo", info), Equals, "/var/snap/name/42/foo")
+	c.Check(builtin.ResolveSpecialVariable("$SNAP_COMMON/foo", info), Equals, "/var/snap/name/common/foo")
+	c.Check(builtin.ResolveSpecialVariable("$SNAP", info), Equals, "/snap/name/42")
+	c.Check(builtin.ResolveSpecialVariable("$SNAP_DATA", info), Equals, "/var/snap/name/42")
+	c.Check(builtin.ResolveSpecialVariable("$SNAP_COMMON", info), Equals, "/var/snap/name/common")
+}
+
+// Check that legacy syntax works and allows sharing read-only snap content
+func (s *ContentSuite) TestConnectedPlugSnippetSharingLegacy(c *C) {
```

Comment:
Both these tests and the test above have duplication which can be easily avoided by using a list of structs defining what's to be tested, instead of copying almost identical logic across lines and tests.